src/btree/bt_split.c

```diff
@@ -594,6 +594,14 @@ __split_parent_discard_ref(WT_SESSION_IMPL *session, WT_REF *ref, WT_PAGE *paren
     WT_IKEY *ikey;
     size_t size;
 
+    /*
```

Comment:
```
         if (next_ref != ref && WT_REF_GET_STATE(next_ref) == WT_REF_DELETED &&
              (btree->type != BTREE_COL_VAR || i != 0) &&
              !F_ISSET_ATOMIC_8(next_ref, WT_REF_FLAG_PREFETCH) &&
              __wti_delete_page_skip(session, next_ref, true) &&
              WT_REF_CAS_STATE(session, next_ref, WT_REF_DELETED, WT_REF_LOCKED)) {
                if (scr == NULL)
                    WT_ERR(__wt_scr_alloc(session, 10 * sizeof(uint32_t), &scr));
                WT_ERR(__wt_buf_grow(session, scr, (deleted_entries + 1) * sizeof(uint32_t)));
                deleted_refs = scr->mem;
                deleted_refs[deleted_entries++] = i;
            }
```

We should check the counter here to not add the ref to the deleted_refs in the first place.

Reply:
@quchenhao Do you mean to move the assert here, or to just check the value?

Reply:
just to check the value